src/EditorFeatures/VisualBasicTest/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.vb

```diff
@@ -1361,6 +1361,51 @@ End Module
             Await TestAsync(markup, expected)
         End Function
 
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)>
+        <WorkItem(32399, "https://github.com/dotnet/roslyn/issues/32399")>
+        Public Async Function TestDontRemoveNecessaryPredefinedCastWithGetType() As Task
```

Comment:
Should we have some casts around DirectCast() to make sure this isn't going to trigger in the case of DirectCast(blah, Object).GetType()? I admit I don't know why somebody would *write* that, but then again I don't understand why somebody would be writing this code either. :smile:

Reply:
Yes, definitely, will update - this will trigger for DirectCast.  Working on it to restrict it only to casts where it is actually doing a predefined / special ctype cast.

Reply:
I've updated it to check for inheritance relationships between the types.  DirectCast and TryCast both require inheritance relationships so this check should rule out those cases.  I've added tests for it as well.  

The subset of CType casts that should not be removed I believe also requires the same relationship as Direct/Try casts, so this covers that case as well.